torch/csrc/cuda/Module.cpp

```diff
@@ -1045,9 +1071,45 @@ static void registerCudaPluggableAllocator(PyObject* module) {
   m.def(
       "_cuda_setCheckpointPoolState",
       [](int device,
-         std::shared_ptr<c10::cuda::CUDACachingAllocator::AllocatorState> pps) {
-        return c10::cuda::CUDACachingAllocator::setCheckpointPoolState(
+         std::shared_ptr<c10::cuda::CUDACachingAllocator::AllocatorState> pps,
+         std::vector<at::Tensor> stale_tensors) {
+        // Could pass in Storage Pointers instead
+        std::unordered_set<c10::StorageImpl*> ptr_set;
+        // iterate on std::vector for determinism
+        std::vector<c10::StorageImpl*> ptrs;
+        for (const auto& ten : stale_tensors) {
+          auto ptr = ten.storage().unsafeGetStorageImpl();
+          if (!ptr_set.count(ptr)) {
+            ptrs.push_back(ten.storage().unsafeGetStorageImpl());
+            ptr_set.insert(ten.storage().unsafeGetStorageImpl());
+          }
+        }
+        auto delta = c10::cuda::CUDACachingAllocator::setCheckpointPoolState(
             device, pps);
+        auto& freed_pointers = delta.ptrs_freed;
+        auto& allocd_pointers = delta.ptrs_allocated;
+
+        std::unordered_set<void*> allocd_set(
+            allocd_pointers.begin(), allocd_pointers.end());
+
+        std::unordered_set<void*> freed_pointer_set;
+        size_t definite_freed_count = 0;
+        for (void* ptr : freed_pointers) {
+          if (!allocd_set.count(ptr)) {
+            definite_freed_count += 1;
+          }
+          freed_pointer_set.insert((ptr));
+        }
+        // that block has already been freed,
+        // so even those this will error, so too will the allcoator
```

Comment:
allocator

Reply:
- allcoator
- correponding
- even those this
- so too will the allcoator